### DISCUSSION TITLE
Security vulnerability alerts

### DISCUSSION THREAD

**Discussion by sapirshloush** [community/community#24766] [Apps, API and Webhooks]
<p>Hi,</p>
<p>I need to update my organization repositories and I have like 130 repos.</p>
<p>I need each of them(the repos) to alert his owners about vulnerability alerts.</p>
<p>There is a way to do this by script? automatically? or the only way is to do this manually?</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hi <a class="mention" href="/sapirshloush">@sapirshloush</a></p>
<p>There isn’t currently an endpoint for switching on Vulnerability Alerts but it is something we’re looking into implementing.</p>
<p>For the moment, enabling Vulnerability Alerts is only available via the UI at the repository level.</p>
<p>We’ll pass your feedback onto the team to make sure they’re aware of your use case. I can’t promise if or when we’ll add an API endpoint for enabling alerts but we’ll make sure the request is in the right hands.</p>
<p>Cheers!</p>

**Reply by jboursier-mwb** [reply]
Hello @AndreaGriffiths11 !

4 years later, do you know if there is an update on this feature? Configuring thousands of repositories is not practical without this API endpoint, and it's a real pain point.

Thanks!

**Reply by roryscarson** [reply]
6 years later, my kingdom for an org wide enablement endpoint. UI workflow per repository when dealing with thousands of repos is a non-starter.

**Comment by farmersmurf** [comment]
<p>Yes very interested in this too, to add some automation into our pipeline and this is much quicker as the scan is already done on commit. Our current process of scanning these in our build pipeline make it quite slow</p>

**Comment by Cieper** [comment]
<p>+1, would also really like to be able to turn on Vulnerability Alerts via the API.</p>

**Comment by iDVB** [comment]
<p>First, it’s surprising that GitHub, an API-first company has this in the UI but not the API.</p>
<p>Secondly, it’s surprising that you have fleshed out, to a great extent, the various <a href="https://developer.github.com/changes/2018-04-24-preview-dependency-graph-and-vulnerability-hooks/" rel="nofollow noopener">advanced APIs around the vulnerabilites</a> but not a single API to simply just enable them?</p>
<p>This feature came out almost a year ago. Many of us need things like this as APIs  in order to automate them. Especially those of us larger clients with 1000+ repos. We can’t turn this feature on manually each time.</p>
<p>Can we get this <strong>fix </strong> bumped up in the roadmap?</p>

**Comment by ThirdEyeRose** [comment]
<p>Hi <a class="mention" href="/idvb">@idvb</a>,</p>
<p>The <a href="https://github.blog/changelog/2019-05-01-sample-of-how-to-enable-security-vulnerability-alerts-for-organizations/" rel="nofollow noopener">endpoint to enable or disable security vulnerabilities was recently released on GitHub.com</a>. It should also make it into GitHub Enterprise in one of the next feature releases.</p>
<p>Hope that helps!</p>

**Comment by larsonreever** [comment]
<p>Dependency Graph API in GraphQL enables you to retrieve information about a repository’s dependency graph. But that’s not all; GH has added a lightweight Repository Vulnerability Alerts API in GraphQL so you can get your security alerts through the API. You can stay up-to-date with the most recent changes using a webhooks that trigger when alerts are created, dismissed, or resolved.</p>
<p>Repository Vulnerability Alerts Webhooks</p>
<p>introduced a new webhook event for repositories called repository_vulnerability_alert. You can get webhooks for create, dismiss, and resolve actions.</p>

**Comment by RAMOTTO** [comment]
<p>Is there any way I can simply erase all changed in my gadget to be typical once more… It’s been a bad dream. Our present procedure of checking these in our fabricate pipeline make it very moderate</p>

**Comment by ThirdEyeRose** [comment]
<p>Hi <a class="mention" href="/ramotto">@ramotto</a>,</p>
<p>I’m sorry, I’m not exactly sure what you’re asking. Could you perhaps rephrase your question?</p>
<p>Thanks! </p>

**Comment by Piedone** [comment]
<p>Do I understand correctly that granting access to security alerts to teams on a repository level (i.e. that you can configure from the UI under repository settings/Security alerts) is not exposed in the API? I understand that enabling security alerts is, as well as you can get notified of alerts via the API but what if I’d like to grant access to the alerts for the team maintaining the repo?</p>
<p>Thank you!</p>

**Comment by vinodkumar4b9** [comment]
<p>Hi Team,</p>
<p>Does some one has the updated Graphql query to fetch the details regarding the vulnerabilities in the repository using graphql queries.</p>

**Comment by nickific** [comment]
<p>{<br>
repository(owner: “nickific”, name: “goof”) {<br>
name<br>
vulnerabilityAlerts(first: 50) {<br>
nodes {<br>
securityAdvisory {<br>
severity<br>
}<br>
}<br>
}<br>
}<br>
}</p>
<p>Alternatively, you can explore your options here: <a href="https://docs.github.com/en/graphql/overview/explorer" class="inline-onebox" rel="noopener nofollow ugc">Explorer - GitHub Docs</a></p>
